### PR TITLE
Support named imports in @Provides annotated methods

### LINKED ISSUE #115: Support named import
The following class is processed successfully by Dagger compiler but not by Anvil
```
package com.squareup.test

import dagger.Module
import dagger.Provides
import java.lang.Runnable as NamedRunnable
        
@Module
object DaggerModule1 {
    @Provides fun provideRunner(): NamedRunnable = NamedRunnable {}
}
```

### PR DESCRIPTION
**Pull request by quiro91**
Fixes #115

### CODE REVIEW

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt [hunk 1]
```diff
@@ -278,6 +278,13 @@ internal fun PsiElement.requireFqName(
         module.findClassOrTypeAlias(importFqName, classReference)?.let { return it.fqNameSafe }
       }
 
+  // Check if it's a named import
```
Comment (vRallev) [inline compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt:281]: Please finish the comment with a dot.

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt [hunk 2]
```diff
@@ -278,6 +278,13 @@ internal fun PsiElement.requireFqName(
...
+  containingKtFile.importDirectives
+    .firstOrNull { classReference == it.importPath?.importedName?.asString() }?.importedFqName
```
Comment (vRallev) [inline compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt:283]: Put `?.importedFqName` on a new line for better readability.

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt [hunk 3]
```diff
@@ -278,6 +278,13 @@ internal fun PsiElement.requireFqName(
...
+    ?.let {
+      module.resolveClassByFqName(it, FROM_BACKEND)?.let { return it.fqNameSafe }
```
Comment (vRallev) [inline compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt:285]: That seems unnecessary. Can't you just return `it` immediately without resolving? You already have the fully qualified name.

**Review by vRallev — CHANGES_REQUESTED** [review]

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt (hunk 3 above)
Comment (quiro91) [reply compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt]: Makes sense, if it matches the reference of the class I can just return it 🤦

**Review by vRallev — APPROVED** [review]

**Pull request merged**